adapters/intertech/intertech.go

```diff
@@ -0,0 +1,326 @@
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: template,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var (
+		requests []*adapters.RequestData
+		errors   []error
+	)
+
+	referer := getReferer(requestData)
+	currency := getCurrency(requestData)
+
+	for i := range requestData.Imp {
+		imp := requestData.Imp[i]
+
+		placementId, err := getPlacementID(imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if err := modifyImp(&imp); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		resolvedUrl, err := a.resolveUrl(*placementId, referer, currency)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		splittedRequestData := splitRequestDataByImp(requestData, imp)
+
+		requestBody, err := json.Marshal(splittedRequestData)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requests = append(requests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     resolvedUrl,
+			Body:    requestBody,
+			Headers: getHeaders(&splittedRequestData),
+			ImpIDs:  openrtb_ext.GetImpIDs(splittedRequestData.Imp),
+		})
+	}
+
+	return requests, errors
+}
+
+func getHeaders(request *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+
+	if request.Device != nil && request.Site != nil {
+		addNonEmptyHeader(&headers, "Referer", request.Site.Page)
+		addNonEmptyHeader(&headers, "Accept-Language", request.Device.Language)
+		addNonEmptyHeader(&headers, "User-Agent", request.Device.UA)
+		addNonEmptyHeader(&headers, "X-Forwarded-For", request.Device.IP)
+		addNonEmptyHeader(&headers, "X-Real-Ip", request.Device.IP)
+		addNonEmptyHeader(&headers, "X-Forwarded-For-IPv6", request.Device.IPv6)
+		headers.Add("Content-Type", "application/json;charset=utf-8")
+		headers.Add("Accept", "application/json")
+	}
+
+	return headers
+}
+
+func addNonEmptyHeader(headers *http.Header, key, value string) {
+	if len(value) > 0 {
+		headers.Add(key, value)
+	}
+}
+
+// splitRequestDataByImp makes a shallow copy of the request for further modification (imp is already a shallow copy)
+func splitRequestDataByImp(request *openrtb2.BidRequest, imp openrtb2.Imp) openrtb2.BidRequest {
+	requestCopy := *request
+	requestCopy.Imp = []openrtb2.Imp{imp}
+
+	return requestCopy
+}
+
+func getPlacementID(imp openrtb2.Imp) (*intertechPlacementID, error) {
+	var ext adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("imp %s: unable to unmarshal ext", imp.ID),
+		}
+	}
+
+	var intertechExt openrtb_ext.ExtImpIntertech
+	if err := json.Unmarshal(ext.Bidder, &intertechExt); err != nil {
```

Comment:
Can you please switch this to using jsonutil? Starting in PBS 3.0 we're switching to our more optimized implementation. Thanks.